packages/peregrine/lib/talons/RootComponents/Category/useCategory.js

```diff
@@ -137,24 +139,43 @@ export const useCategory = props => {
         // applied filters. Follow-up in PWA-404.
         newFilters['category_id'] = { eq: String(id) };
 
-        runQuery({
-            variables: {
-                currentPage: Number(currentPage),
-                id: Number(id),
-                filters: newFilters,
-                pageSize: Number(pageSize),
-                sort: { [currentSort.sortAttribute]: currentSort.sortDirection }
+        if (!isEqual(filtersRef.current, newFilters)) {
+            filtersRef.current = newFilters;
+        }
+    }, [filterTypeMap, id, pageSize, search]);
+
+    // Query with filters and whenenver they change.
+    useEffect(
+        () => {
+            if (filtersRef.current) {
+                console.log('filtersRef.current', filtersRef.current);
+                runQuery({
+                    variables: {
+                        currentPage: Number(currentPage),
+                        id: Number(id),
+                        filters: filtersRef.current,
+                        pageSize: Number(pageSize),
+                        sort: {
+                            [currentSort.sortAttribute]:
+                                currentSort.sortDirection
+                        }
+                    }
+                });
             }
-        });
-    }, [
-        currentPage,
-        currentSort,
-        filterTypeMap,
-        id,
-        pageSize,
-        runQuery,
-        search
-    ]);
+        },
+        // exhaustive-deps complains about refs...
+        /* eslint-disable react-hooks/exhaustive-deps */
```

Comment:
Is there a better way to solve this without a ref so we avoid having to disable the lint rule?